pillarbox-player/src/main/java/ch/srgssr/pillarbox/player/qos/QoSCoordinator.kt

```diff
@@ -16,40 +16,56 @@ import androidx.media3.exoplayer.ExoPlayer
 import androidx.media3.exoplayer.analytics.AnalyticsListener
 import androidx.media3.exoplayer.source.LoadEventInfo
 import androidx.media3.exoplayer.source.MediaLoadData
+import ch.srgssr.pillarbox.player.BuildConfig
 import ch.srgssr.pillarbox.player.analytics.PillarboxAnalyticsListener
 import ch.srgssr.pillarbox.player.analytics.PlaybackSessionManager
 import ch.srgssr.pillarbox.player.analytics.metrics.MetricsCollector
 import ch.srgssr.pillarbox.player.analytics.metrics.PlaybackMetrics
 import ch.srgssr.pillarbox.player.extension.getPositionTimestamp
+import ch.srgssr.pillarbox.player.network.PillarboxHttpClient
 import ch.srgssr.pillarbox.player.qos.models.QoETimings
 import ch.srgssr.pillarbox.player.qos.models.QoSError
 import ch.srgssr.pillarbox.player.qos.models.QoSEvent
 import ch.srgssr.pillarbox.player.qos.models.QoSEvent.StreamType
 import ch.srgssr.pillarbox.player.qos.models.QoSMedia
 import ch.srgssr.pillarbox.player.qos.models.QoSMessage
+import ch.srgssr.pillarbox.player.qos.models.QoSMessage.EventName
+import ch.srgssr.pillarbox.player.qos.models.QoSMessageData
 import ch.srgssr.pillarbox.player.qos.models.QoSSession
 import ch.srgssr.pillarbox.player.qos.models.QoSStall
 import ch.srgssr.pillarbox.player.qos.models.QoSTimings
 import ch.srgssr.pillarbox.player.runOnApplicationLooper
 import ch.srgssr.pillarbox.player.utils.DebugLogger
 import ch.srgssr.pillarbox.player.utils.Heartbeat
+import kotlinx.coroutines.CoroutineScope
 import java.io.IOException
+import java.net.URL
 import kotlin.coroutines.CoroutineContext
 import kotlin.time.Duration.Companion.ZERO
+import kotlin.time.Duration.Companion.milliseconds
 import kotlin.time.Duration.Companion.seconds
 
 internal class QoSCoordinator(
     private val context: Context,
     private val player: ExoPlayer,
     private val metricsCollector: MetricsCollector,
-    private val messageHandler: QoSMessageHandler,
     private val sessionManager: PlaybackSessionManager,
     private val coroutineContext: CoroutineContext,
 ) : PillarboxAnalyticsListener,
     MetricsCollector.Listener,
     PlaybackSessionManager.Listener {
     private val window = Timeline.Window()
 
+    var messageHandler: QoSMessageHandler = if (BuildConfig.DEBUG) {
+        RemoteQoSMessageHandler(
+            httpClient = PillarboxHttpClient(),
+            endpointUrl = URL("https://httpbin.org/post"),
+            coroutineScope = CoroutineScope(coroutineContext),
+        )
+    } else {
+        LogcatQoSMessageHandler()
+    }
```

Comment:
Once we have the final URL of the QoS server, we can replace this with:
```suggestion
    var messageHandler: QoSMessageHandler = RemoteQoSMessageHandler(
        httpClient = PillarboxHttpClient(),
        endpointUrl = URL("<qos_url>"),
        coroutineScope = CoroutineScope(coroutineContext),
    )
```